### ISSUE TITLE
Can Pygame benefit from PEP 703 (making the GIL optional)?

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [pygame/pygame#3934]
Hi!

I'm Gabriel de Marmiesse and I'm helping Sam Gross to find out if maintainers of selected Python packages would benefit from PEP703.

If you don't know what is PEP 703, it's about making the global interpreter lock (GIL) optional.
Long story short, there would be a ~10% performance impact on single threaded programs, but would allow multithreaded programms to take full advantage of multiple cores.
[You can play with it here](https://github.com/colesbury/nogil).

If you have the time, we would like to know is if the maintainers of Pygame would benefit from PEP 703 or not. There are multiple situation Pygame could be in.

1) Pygame already uses threads and PEP 703 will likely make some features faster by allowing parallelization of cpu computation without any change.
2) Pygame doesn't use threads, but PEP 703 would allow you to rewrite some features with multithreading, thus making it faster.
3) Pygame uses multithreading with a low-level language (rust, C, C++...) or multiprocessing and using PEP 703 would allow a rewrite and lower the maintenance burden.
4) There is no computation that can be parallelized.
5) Pygame wouldn't use multithreading, but it's very likely that users will call this package functions and classes from multiple threads because the workload can often be parralelized.

The Python language could never use multiple cores with multithreading, so it's hard to imagine what the future of the language would look like if we were to enable it. Feel free to tell us if you see new potential use cases with this PEP. Thanks for your time!

If you want to know more about PEP 703:
* [The PEP](https://peps.python.org/pep-0703/)
* [Python 3.9 implementation with install instructions](https://github.com/colesbury/nogil)
* [Python 3.12 implementation](https://github.com/colesbury/nogil-3.12)
* [Discussion about PEP 703](https://discuss.python.org/t/pep-703-making-the-global-interpreter-lock-optional-3-12-updates/26503)
* [Discussion about the multicore future of Python](https://discuss.python.org/t/a-fast-free-threading-python/27903)

**Comment by gabrieldemarmiesse** [comment]
Thank you for the feedback! 

To answer the question, the PEP doesn't list this directly (@colesbury maybe this could be an improvement to the pep? I know the PEP is saying that an HOWTO guide will be made, but maybe it's worth doing it earlier if some of your time is available?).
A good reference concerning the changes to be made by the extensions maintainers would be this presentation at EuroPython:
https://youtu.be/9OOJcTp8dqE?t=1889

It has some good examples about what was needed to change in some popular scientific packages.
Long story short, the work might not be as significant as you might imagine. I believe most of the work would actually come from supporting a double distribution of Python for some years, but not from putting locks everywhere in your code. But that's only my take on the subject. I might be very wrong.
